static/app/stores/persistedStore.tsx

```diff
@@ -0,0 +1,121 @@
+import React, {
+  createContext,
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+
+import useApi from 'sentry/utils/useApi';
+import useOrganization from 'sentry/utils/useOrganization';
+import {OnboardingState} from 'sentry/views/onboarding/targetedOnboarding/types';
+
+import OrganizationStore from './organizationStore';
+import {useLegacyStore} from './useLegacyStore';
+
+type PersistedStore = Readonly<{
+  onboarding: Partial<OnboardingState> | null;
```

Comment:
The way I'm typing it here requires that each time we add a category, we need to go in here and add the type to `PersistedStore`. Should be fine, given that we also need to add the category name on the backend, but i'll be better if we can define types in the files that they're actually used.

Reply:
Yeah, I like using keys with usePersistedStoreCategory<Key> - like we discusses yesterday. This has less chance of falling out of sync that usePersistedStore<T> where T can be any object. This way it'll be easier to maintain and update in the long term